### DISCUSSION TITLE
Is it possible to control the page numbering?

### DISCUSSION THREAD

**Discussion by KevinCT** [QuestPDF/QuestPDF#281] [Q&A] [answered]
Hello!

I have a scenario where I would like to generate a PDF consisting of multiple documents. I am able to achieve this by creating a separate "Page" for each document. The issue I have yet to solve is the possibility of resetting the page numbering for each document.

For example, adding two documents with 3 pages each, I would get the following printed as the page numbering: 1/6, 2,/6, 3/6 ... 6/6.
I would like the page numbering to reset for each document so that I get the following: 1/3, 2,/3, 3/3, 1/3, 2/3, 3/3.

It would be helpful, if there is a way to achieve this since it would open up the possiblity of merging multiple documents into one pdf without the need of an external library.

I would also like to give a sincere thanks for developing and maintaining such an awesome library!

```
public void Compose(IDocumentContainer container)
{
   // Document 1
    container
              .Page(page =>
              {
                  //Code omitted
                  page.Footer().AlignCenter().Text(x =>
                  {
                      x.CurrentPageNumber();
                      x.Span(" / ");
                      x.TotalPages();
                  });
              });
  
    // Document 2
    container
              .Page(page =>
              {
                  //Code omitted
                  page.Footer().AlignCenter().Text(x =>
                  {
                      x.CurrentPageNumber();   <---- I would like the page numbering here to start from one again.
                      x.Span(" / ");
                      x.TotalPages();
                  });
              });
}
```

**Comment by sasaPantic** [comment, marked as answer]
Look at here https://www.questpdf.com/api-reference/text.html#sections

**Comment by DawidRobertNowak** [comment]
Hello @KevinCT,

We are closing this thread due to inactivity. We apologize for the delayed response.

As @sasaPantic pointed out, this is now achievable using the Sections feature. You can wrap each document's content in a named `.Section()`, and then use `PageNumberWithinSection()` and `TotalPagesWithinSection()` in the footer to get per-document page numbering that resets for each section.

You can find the full documentation and examples here:
https://www.questpdf.com/api-reference/text/page-numbers.html

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
